src/core/Suite.js

```diff
@@ -88,14 +88,14 @@ getJasmineRequireObj().Suite = function(j$) {
 
   Suite.prototype.sharedUserContext = function() {
     if (!this.sharedContext) {
-      this.sharedContext = this.parentSuite ? clone(this.parentSuite.sharedUserContext()) : {};
+      this.sharedContext = this.parentSuite ? this.parentSuite.clonedSharedUserContext() : new j$.UserContext();
     }
 
     return this.sharedContext;
   };
 
   Suite.prototype.clonedSharedUserContext = function() {
-    return clone(this.sharedUserContext());
+    return extend(new j$.UserContext(), this.sharedUserContext());
```

Comment:
Now that there is a place to put it, could this be a method on the `UserContext` constructor, so Suite would call something like:

```javascript
j$.UserContext.fromExisting(this.sharedUserContext())
```

And remove all of the `extend` stuff from Suite.